Do nothing when the same block is included again

mjtko opened the pull request:
If the same block is included multiple times, we no longer raise an exception or overwrite the included block instance variable.

Fixes #14802.

mjtko commented:
@simi: Please review the discussion on #14802 for context.

Regarding (2) - it depends what you mean by "work".  I don't think it works -- I've reopened a discussion about that aspect of this change on #14802, can you add your view there please?

gmcgibbon commented on activesupport/test/concern_test.rb:
```diff
@@ -128,4 +128,12 @@ def test_raise_on_multiple_included_calls
+  def test_no_raise_on_same_included_call
+    assert_nothing_raised do
+      2.times do
+        load File.expand_path("../fixtures/some_module.rb", __FILE__)
```
I think you can do `require_relative` here, right?

mjtko replied:
Using `require_relative` rather than `load` would result in a no-op on the second iteration, no?

gmcgibbon commented on activesupport/test/fixtures/some_module.rb:
```diff
@@ -0,0 +1,11 @@
+  extend ActiveSupport::Concern
+
+  included do
+    # shouldn't raise
```
This is misleading. I would adjust the comment to say `# shouldn't raise when module is required more than once`. I would also consider renaming the module to `SomeConcern`

mjtko replied:
Happy with the renaming and I'll expand the comment tho, as above, I think it needs to be `when module loaded more than once` rather than `...required...`.

rafaelfranca commented:
Can you squash your commits?

mjtko commented:
@rafaelfranca: yup, was just going to wait until they were all done first, but can do that now, np.

gmcgibbon commented:
Is the first commit necessary?

mjtko commented:
No, I just wanted to ensure the original author of the patch, @razum2um, received sufficient credit.

mjtko committed:
Do nothing when the same block is included again.

If the same block is included multiple times, we no longer raise an exception
or overwrite the included block instance variable.

Fixes #14802.

[Mark J. Titorenko + Vlad Bokov]

mjtko commented:
Ok, thanks for the example! Squashed and modified accordingly. 😄

kalashnikovisme commented:
could you please tell me, how should I learn what Rails release has this fix?

razum2um commented:
@mjtko appreciate the credit, thanks 👍 sorry that I had to leave earlier PR & discussion.
long live rails community ❤️
